great_expectations/experimental/metric_repository/metrics.py

```diff
@@ -42,11 +53,12 @@ def __new__(cls, *args, **kwargs):
         instance = super().__new__(cls)
         return instance
 
-    id: uuid.UUID = Field(description="Metric id")
```

Comment:
I removed the id fields on MetricRun and Metric since they will be generated by the backend and don't yet have a use here in OSS. If in the future we retrieve metrics with IDs we can add this back in as an optional field.